java/core/src/test/org/apache/orc/impl/TestRecordReaderImpl.java

```diff
@@ -403,16 +406,18 @@ public void testGetMin() throws Exception {
 
   private static OrcProto.ColumnStatistics createIntStats(Long min,
                                                           Long max) {
+    OrcProto.ColumnStatistics.Builder result =
+        OrcProto.ColumnStatistics.newBuilder();
     OrcProto.IntegerStatistics.Builder intStats =
         OrcProto.IntegerStatistics.newBuilder();
     if (min != null) {
       intStats.setMinimum(min);
+      result.setNumberOfValues(1);
```

Comment:
I am not sure why this is needed? Also if min != max wouldn't we have > 1 values?
PSA: I am not familiar with the Orc codebase, especially the protobuf parts. So may be totally missing the point.

Reply:
Ok, the fix was about handling ColumnStatistics when there were no non-null values. So the added branch checks the number of values in the statistics. In real files, the only way to have a minimum or maximum is to have had at least one value, so it works.

In the sarg tests though, we were creating artificial ColumnStatistics that had minimum and maximum without any values. They broke with the new code. The fix was to make the artificial ColumnStatistics have a non-zero number of values.